### PR TITLE
builtin: Implement builtin sum

### PR DESCRIPTION
**Pull request by corona10**
Implement built-in sum.

### CODE REVIEW

**Comment by corona10** [comment]
@ncw PTAL

File: builtin/tests/builtin.py [hunk 1]
```diff
@@ -159,6 +159,30 @@ class C: pass
...
+
+try:
+    sum([1,2,3], 'hi')
+except TypeError as e:
+    if e.args[0] != "sum() can't sum strings [use ''.join(seq) instead]":
+        raise
+    ok = True
+assert ok, "TypeError not raised"
+
+try:
```
Comment (sbinet) [inline builtin/tests/builtin.py]: perhaps also add a test for:
```python
sum(['h', 'i'])
```

?

**Comment by corona10** [comment]
@sbinet I've updated PTAL

**Comment by corona10** [comment]
@ncw Can you take a look?

**Comment by corona10** [comment]
@ncw 
Your CL is awesome!
I 've rebased the master and fix a code which is related to failing.
PTAL

**Comment by ncw** [comment]
Look great now - thanks :-)

**Pull request merged**